### ISSUE TITLE
Show cluster bounds

### ISSUE THREAD

**Issue by danielhers** [hasadna/anyway#184]
When the cursor is hovering over a cluster, show its bounds.

example:
https://www.mapbox.com/mapbox.js/example/v1.0.0/leaflet-markercluster/

**Comment by robi-y** [comment]
This is probably a leaflet feature - I think we will need to switch from
google maps in order to use it

**Comment by danielhers** [comment]
It was just an example. Maybe we can replicate this. @michalten

Op ma 23 feb. 2015 21:49 schreef robi-y <notifications@github.com>:

> This is probably a leaflet feature - I think we will need to switch from
> google maps in order to use it
>
> On Mon, Feb 23, 2015 at 6:53 PM, Daniel Hershcovich <

**Comment by samuelregev** [comment]
google maps knows how to mark an area
![image](https://cloud.githubusercontent.com/assets/5688494/8030600/cdd02608-0dd0-11e5-9209-ec291d752680.png)

so i'm guessing it's possible

**Comment by omerxx** [comment]
The cluster is calculated in a circle shaped parameter. We could show the round parameter...

**Comment by pint12** [comment]
hi .
this can be achieved also by using CSS3 :hover commend on the "cluster" class .
also ill recommend to remove all the inline style !!

**Comment by omerxx** [comment]
Go for it..
What's with the inline?

**Comment by pint12** [comment]
i thought it is written within the HTML 
but now i see it is been granted by a plugin MarkerClustererPlus v2.0 
who implemented it ? 
i see in the docs they they support 
mouseover	c:Cluster	This event is fired when the mouse moves over a cluster marker.

**Comment by pint12** [comment]
done ! 
i add to css 
.cluster:hover div:first-child  {
  position: absolute;
    top: 0px;
    left: 0px;
    color: black;
    font-weight: bold;
    font-size: 15px;
    font-family: Alef, sans-serif;
    border: 10px solid blue !important;
    border-radius: 50%;
    text-align: center;
    width: 32px;
    line-height: 32px;
    vertical-align: middle;
    background-color: white;
    opacity: 0.7;
      border: 10px solid blue;
}

now when you are going over the round it is becoming blue !

**Comment by danielhers** [comment]
Great! Please open a pull request.

**Comment by danielhers** [comment]
@pint12 you should open a pull request to `hasadna/anyway` (`dev` branch).

**Comment by pint12** [comment]
i think i did now can you check ?

**Comment by danielhers** [comment]
I don't see it.

**Comment by danielhers** [comment]
@pint12 I think you didn't push to your branch yet. You should push to https://github.com/pint12/anyway/tree/dev and then create a request to merge it into `dev` on `hasadna/anyway`. What you did [here](https://github.com/pint12/anyway/pull/1) is the other way around...

**Comment by omerxx** [comment]
@pint12 let me know if you need any git help
You're invited to the slack feel free to IM me

**Comment by danielhers** [comment]
@pint12 I merged your change on my computer and looked at the result, but it's not quite what was intended in this issue. What I see is this:
![selection_001](https://cloud.githubusercontent.com/assets/263363/10664292/d16e3f10-78ca-11e5-84f6-841eb5d9ca34.png)

While the issue is to delineate the area on the map that the cluster represents, i.e. a polygon containing all points in the cluster.

**Comment by pint12** [comment]
@danielhers  i have tried to that but then i saw you are using a built JS lib MarkerClustererPlus v2.0 . 
this lib does not have the option for creating polygon clusrures . 
we can cosider using another Js lib but implementation might be painfull .
what do think ?

**Comment by danielhers** [comment]
Yes, that's why the borders feature hadn't been implemented so far.
You're welcome to try switching us to a different JS library that does
support polygon borders, if there's no way to do it with our current
library.

**Comment by pint12** [comment]
how this is the use with the cluster lib? who implemented it ?

**Comment by danielhers** [comment]
Mainly @njenia, I think...

**Comment by pint12** [comment]
ok i investigated it a bit more. 
in order to create a polygon (with google )
i need all the  lan and lat points that exists in cluster 
for exmple 
triangleCoords = [
    {lat: 25.774, lng: -80.190},
    {lat: 18.466, lng: -66.118},
    {lat: 32.321, lng: -64.757},
    {lat: 25.774, lng: -80.190}
  ];

but in current version i am getting a big array with only the center postion 
{"clusters":
 [{"latitude": 32.0898447, "longitude": 34.819303699999, "size": 6}, 
 {"latitude": 32.0903437, "longitude": 34.8256844, "size": 13}]

this position is from backend? if yes can it be changed?

**Comment by danielhers** [comment]
Yes, the whole idea in sending the clusters is that it's much less data so it requires less traffic and less client computation.
You can change the server-side clustering code to send the bounding polygon's corners instead.

**Comment by danielhers** [comment]
The clusters are calculated in https://github.com/hasadna/anyway/blob/dev/clusters_calculator.py

**Comment by pint12** [comment]
i think cluster calculator should send back  the bounding polygon's corners instead in addition to current data... 
i dont know python (yet) .can you assist?

**Comment by danielhers** [comment]
Yes, that's what I said.
You can learn Python using any online tutorial such as [this](http://www.learnpython.org/).
You can learn about the Web framework we are using (Flask) [here](http://www.fullstackpython.com/).
Once you are comfortable with the language, please come to the development meetings and you will surely get help with the task at hand.
